### PR TITLE
SOC absorption for TDDFPT

### PR DESCRIPTION
**Pull request by Tarandin**
We, @annahehn and I, have implemented a probational SOC correction for the tddfpt2 module. Basically, we changed the sums used in the xas_tdp module to use all Orbitals. We checked our implementation against a benchmark used by Christel M. Marian et al. (https://doi.org/10.1021/acs.jctc.6b01122) with reasonable deviations for 11 out of 14 small molecules. For the remaining three, dithiin, furan and quinoxaline, the corresponding excitations could not be identified.

### CODE REVIEW

File: src/input_cp2k_properties_dft.F [hunk 1]
```diff
@@ -1603,6 +1603,31 @@ SUBROUTINE create_tddfpt2_section(section)
...
       CALL section_release(subsection)
 
+      ! SOC functional
+
+      CALL section_create(subsection, __LOCATION__, name="SOC", &
+                          description="Is jet to be implemented", &
+                          n_keywords=2, n_subsections=0, repeats=.FALSE.)
+
+      CALL keyword_create(keyword, __LOCATION__, name="SINGLETT", &
+         description="Number of singlett states", &
```
Comment (mkrack) [inline src/input_cp2k_properties_dft.F]: I think `singlet` and `triplet` is the correct spelling.

File: src/cp_control_types.F [hunk 2]
```diff
@@ -512,6 +512,12 @@ MODULE cp_control_types
       !> user-defined reference point
       REAL(kind=dp), DIMENSION(:), POINTER :: dipole_ref_point
       !
+      ! SOC subsection
+      ! Number of Singlett states to be used
+      INTEGER                              :: soc_singlett
+      ! Number of triplett states to be used
+      INTEGER                              :: soc_triplett
+      !
```
Comment (mkrack) [inline src/cp_control_types.F:517]: soc_singlet soc_triplet

File: src/input_cp2k_properties_dft.F (hunk 1 above)
Comment (Tarandin) [reply src/input_cp2k_properties_dft.F]: Thank you very much, you are absolutely right.  This error should be resolved now.

File: src/cp_control_types.F (hunk 2 above)
Comment (Tarandin) [reply src/cp_control_types.F:517]: Thank you. This error should be resolved now.

**Comment by oschuett** [comment]
Shall we (squash) merge this?

**Pull request merged**